packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -0,0 +1,474 @@
+/// the `initializeMessages()` function and the per-locale backing store for `Intl.message()`.
+/// The message catalog is produced by an `intl` tool that analyzes the source code for
+/// classes that contain `Intl.message()` calls. In this case that would just be the
+/// `MyLocalizations` class.
+///
+/// One could choose another approach for loading localized resources and looking them up while
+/// still conforming to the structure of this example.
+class Localizations extends StatefulWidget {
+  Localizations({
+    Key key,
+    @required this.locale,
+    this.delegate,
+    this.child
+  }) : super(key: key) {
+    assert(locale != null);
+  }
+
+  /// The resources returned by [Localizations.of] will be specific to this locale.
+  final Locale locale;
+
+  /// This delegate is responsible for loading and caching resources for `locale`.
+  ///
+  /// Typically defined with [DefaultLocalizationsDelegate].
+  final LocalizationsDelegate delegate;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  /// The locale of the Localizations widget for the widget tree that
+  /// corresponds to [BuildContext] `context`.
+  static Locale localeOf(BuildContext context) {
+    assert(context != null);
+    final _LocalizationsScope scope = context.inheritFromWidgetOfExactType(_LocalizationsScope);
+    return scope.localizedResourcesState.locale;
+  }
+
+  /// Returns the 'type' localized resources for the widget tree that
+  /// corresponds to [BuildContext] `context`.
+  ///
+  /// This method is typically used by a static factory method on the 'type'
+  /// class. For example Flutter's MaterialLocalizations class looks up Material
+  /// resources with a method defined like this:
+  ///
+  /// ```dart
+  /// static MaterialLocalizations of(BuildContext context) {
+  ///    return Localizations.of<MaterialLocalizations>(context, MaterialLocalizations);
+  /// }
+  /// ```
+  static T of<T>(BuildContext context, Type type) {
+    assert(context != null);
+    assert(type != null);
+    final _LocalizationsScope scope = context.inheritFromWidgetOfExactType(_LocalizationsScope);
+    return scope.localizedResourcesState.resourcesFor<T>(type);
+  }
+
+  @override
+  _LocalizationsState createState() => new _LocalizationsState();
+}
+
+class _LocalizationsState extends State<Localizations> {
+  final GlobalKey _localizedResourcesScopeKey = new GlobalKey();
+
+  Locale get locale => _locale;
+  Locale _locale;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.delegate == null)
+      _locale = widget.locale;
+    else
+      load(widget.locale);
+  }
+
+  @override
+  void didUpdateWidget(Localizations old) {
+    super.didUpdateWidget(old);
+    if (widget.delegate != old.delegate && widget.delegate != null)
+      load(widget.locale);
+  }
+
+  void load(Locale locale) {
+    final LocalizationsDelegate delegate = widget.delegate;
+    if (delegate == null) {
```

Comment:
Yes, if the WidgeApp's Localizations widget is built with a new locale but no delegate. The tests do this.  It would happen in WidgetsApp with no localized resources if the user changed the system locale.